plugins/plugin_utils/unsafe.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2023, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+
+from ansible.module_utils.six import binary_type, text_type
+from ansible.module_utils.common._collections_compat import Mapping, Set
+from ansible.module_utils.common.collections import is_sequence
+from ansible.utils.unsafe_proxy import (
+    AnsibleUnsafe,
+    wrap_var as _make_unsafe,
+)
+
+_RE_TEMPLATE_CHARS = re.compile(u'[{}]')
+_RE_TEMPLATE_CHARS_BYTES = re.compile(b'[{}]')
+
+
+def make_unsafe(value):
+    if value is None or isinstance(value, AnsibleUnsafe):
+        return value
+
+    if isinstance(value, Mapping):
+        return dict((make_unsafe(key), make_unsafe(val)) for key, val in value.items())
+    elif isinstance(value, Set):
+        return set(make_unsafe(elt) for elt in value)
+    elif is_sequence(value):
+        return type(value)(make_unsafe(elt) for elt in value)
+    elif isinstance(value, binary_type):
+        if _RE_TEMPLATE_CHARS_BYTES.search(value):
+            value = _make_unsafe(value)
+        return value
+    elif isinstance(value, text_type):
+        if _RE_TEMPLATE_CHARS.search(value):
+            value = _make_unsafe(value)
+        return value
```

Comment:
I was tempted to say "bleh, regex, use `something in value`, but a quick `timeit` says regex is actually faster:

```python
import re
import timeit

binary_type = bytes
text_type = str

_RE_TEMPLATE_CHARS = re.compile(u'[{}]')
_RE_TEMPLATE_CHARS_BYTES = re.compile(b'[{}]')

def _make_unsafe(value):
    return value

def make_unsafe(value):
    if isinstance(value, binary_type):
        if _RE_TEMPLATE_CHARS_BYTES.search(value):
            value = _make_unsafe(value)
        return value
    elif isinstance(value, text_type):
        if _RE_TEMPLATE_CHARS.search(value):
            value = _make_unsafe(value)
        return value
    return value

def make_unsafe_in(value):
    if isinstance(value, binary_type):
        if any(check in value for check in [b'{{', b'}}']):
            value = _make_unsafe(value)
        return value
    if isinstance(value, text_type):
        if any(check in value for check in ['{{', '}}']):
            value = _make_unsafe(value)
        return value
    return value

print(timeit.timeit('make_unsafe("adsdfsdf{{sdfsdfsf}}aaaaa")', globals=globals()))
print(timeit.timeit('make_unsafe(b"adsdfsdf{{sdfsdfsf}}aaaaa")', globals=globals()))
print(timeit.timeit('make_unsafe_in("adsdfsdf{{sdfsdfsf}}aaaaa")', globals=globals()))
print(timeit.timeit('make_unsafe_in(b"adsdfsdf{{sdfsdfsf}}aaaaa")', globals=globals()))
```

```console
% python3.12 /tmp/bench.py
0.27897425496485084
0.2632201029919088
0.6106972509878688
0.8229789500473998
```

of course, PyPy is a magnitude faster, but still regex wins:
```console
% pypy3.9 /tmp/bench.py
0.026400349976029247
0.018924177973531187
0.05259729997487739
0.047663011006079614
```

Reply:
But once I switch from `any(…)` to an explicit `'{{' in value or '}}' in value`, things get faster:
```console
% python3.12 /tmp/bench.py
0.29455614904873073
0.2644505600328557
0.10402327799238265
0.28959875099826604

% pypy3.9 /tmp/bench.py  
0.023167914012447
0.022700645960867405
0.0018618669710122049
0.0015594420256093144
```

Reply:
The above is btw with:
```
% rpm -q python3 pypy3.9
python3-3.12.2-2.fc39.x86_64
pypy3.9-7.3.15-2.3.9.fc39.x86_64
```